### PR TITLE
remove 3.7 support

### PR DESCRIPTION
**Pull request by xiki-tempula**
Remove the support for py3.7

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dependencies:
 - python
 - numpy
 - pandas
-- pymbar
+- pymbar >=3.0.5,<4
```
Comment (orbeckst) [inline environment.yml:8]: oops, ... must have forgotten this earlier

File: src/alchemlyb/__init__.py [hunk 2]
```diff
@@ -46,15 +46,6 @@ def concat(objs, *args, **kwargs):
...
-        attrs = objs[0].attrs
-    except IndexError: # except empty list as input
-        raise ValueError('No objects to concatenate')
-
-    for obj in objs:
-        if attrs != obj.attrs:
-            raise ValueError('All pandas objects should have the same attrs.')
-    new = pd.concat(objs, *args, **kwargs)
-    new.attrs = attrs
-    return new
```
Comment (orbeckst) [inline src/alchemlyb/__init__.py:60]: keep the original code, don't replace with pd.concat

Comment (orbeckst) [reply src/alchemlyb/__init__.py:60]: (In another discussion on MDAnalysis, @IAlibay convinced me that we should just preserve the deprecated code exactly as is; also minimizes the diff)

File: src/alchemlyb/__init__.py [hunk 3]
```diff
@@ -46,15 +46,6 @@ def concat(objs, *args, **kwargs):
...
+    warn('This method will be deprecated in 1.0.0. Use pd.concat instead.',
```
Comment (orbeckst) [inline src/alchemlyb/__init__.py:49]: Fix text:

This function will me removed in 1.0.0. Use pandas.concat() instead.

File: readthedocs.yml [hunk 4]
```diff
@@ -8,7 +8,7 @@ formats:
   - pdf
 
 python:
-  version: 3.7
+  version: 3.8
```
[the remark this answers is not in the document]
Comment (xiki-tempula) [reply readthedocs.yml]: Error
Problem in your project's configuration. Invalid "python.version": expected one of (2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, pypy3.5), got 3.9

**Comment by xiki-tempula** [comment]
@orbeckst 
I'm also a bit reluctant to remove the check and we shall keep the alchemlyb.concat.
I'm happy to drop the support of py3.7.

**Review by orbeckst — CHANGES_REQUESTED** [review]
please add CHANGELOG entry, otherwise looking good

**Comment by xiki-tempula** [comment]
@orbeckst Done.

**Review by orbeckst — APPROVED** [review]

**Pull request merged**